ewoms/common/simulator.hh

```
@@ -135,11 +153,27 @@ public:
 
         if (verbose_)
             std::cout << "Finish init of the model\n" << std::flush;
-        model_->finishInit();
+        try {
+            model_->finishInit();
+        }
+        catch (const std::exception& e) {
+            exceptionThrown = 1;
+        }
+        if (cc.max(exceptionThrown) == 1) {
+            throw std::runtime_error("Could not finish init of the model.");
```

you probably need to call `MPI_Finalize()` here before you throw, else you will get undefined behavior due to some processes being terminated while others still want to communicate.